### DISCUSSION TITLE
Input Custom Masses and Period

### DISCUSSION THREAD

**Discussion by nbungart1999** [phoebe-project/phoebe2#451] [Questions]
How do I go about setting the value of the masses and period of a default binary? I have gone through the tutorial but the only example I could find is when they set the mass of only one star. I know it's not straightforward because when I use b.set_value('mass', component='primary', value=1) it returns an error because it is a constrained parameter.

**Comment by gecheline** [comment]
The masses and period are constrained through Kepler’s third law, which also includes the mass ratio and semi-major axis of the binary. You need to “flip” those constraints first, so that you can set the masses and constrain the other parameters (for example, mass ratio and semi-major axis since you have a value for the period too).

You can do something like this:
`b.flip_constraint('mass@primary', solve_for='sma@binary')
b.flip_constraint('mass@secondary', solve_for='q')

b.set_value('mass', component='primary', value=1)
b.set_value('mass', component='secondary', value= 0.8)`

**Reply by nbungart1999** [reply]
Oh okay, I'm still learning so it's basically like the system needs to have a certain amount of constraints so in order to set the masses you need to give back two values to take the spots for those constraints? Otherwise, I understand your solution.

**Reply by gecheline** [reply]
Yes, not all parameters in the PHOEBE bundle are independent and constraints are there to make sure that when you update a  free parameter, all the others that depend on it are updated accordingly. So, when you set the masses, you need to tell PHOEBE which parameters to update instead through Kepler's 3rd law, and which to leave "free". You can see more in the constraints tutorials here: http://phoebe-project.org/docs/2.3/tutorials/constraints
